tests/actors.rs

```diff
@@ -0,0 +1,143 @@
+use crate::common::setup_test_environment;
+use sqlx::{Pool, Postgres};
+
+mod common;
+
+// We use sqlx::test so sqlx will create a database instance for the test,
+// set it up based on the migration data, and shut it down upon success.
+// The pool passed to the function is that created database.
+#[sqlx::test]
+async fn get_actors(pool: Pool<Postgres>) {
+    let app = setup_test_environment(&pool, "actors").await;
+
+    // No actors
+    let response = app.client.get(&app.url).send().await.unwrap();
+    assert_eq!(response.status().as_u16(), 200);
+    let content = response.text().await.unwrap();
+    assert_eq!(content, "[]");
+
+    // Single actor
+    // Insert a value into the empty db so we have something to get
+    sqlx::query!("INSERT INTO actors (name) VALUES ($1)", "TestName")
+        .execute(&pool)
+        .await
+        .unwrap();
+
+    let response = app.client.get(&app.url).send().await.unwrap();
+    assert_eq!(response.status().as_u16(), 200);
+    let expected = r#"[{"id":1,"name":"TestName"}]"#;
+    let content = response.text().await.unwrap();
+    assert_eq!(content, expected);
+
+    // Many actors
+    for n in 2..5 {
+        sqlx::query!("INSERT INTO actors (name) VALUES ($1)", n.to_string())
+            .execute(&pool)
+            .await
+            .unwrap();
+    }
+
+    let response = app.client.get(&app.url).send().await.unwrap();
+    assert_eq!(response.status().as_u16(), 200);
+    let expected = r#"[{"id":1,"name":"TestName"},{"id":2,"name":"2"},{"id":3,"name":"3"},{"id":4,"name":"4"}]"#;
+    let content = response.text().await.unwrap();
+    assert_eq!(content, expected);
```

Comment:
Cool idea to put the insert in a loop.  

Makes me think that we could optionally tighten this up a bit more now that that's there. Supposed we had the insert and the get/assert in each loop iteration:
``` rust
let app = setup_test_environment(&pool, "actors").await;

for n in 0..3 {
    sqlx::query!("insert");
    let response = app.client.get();
    let response: HashMap<String, Value> = serde_json::from_str(response).unwrap();
    assert_eq!(response.len(), n);
    
    if n == 0 {
        assert_eq!(response.get(n), "[]");
    } else {
        assert_eq!(response.get(n), format!(r#"{"id":1,"name":"{}"}#"), n);
    }
}
```
It would cover 0, 1, and many all in one loop

Reply:
Found the serde_json line here https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=d26b586a27c73dd2d11e3669d9bb203c

Reply:
I feel like that makes it a bit too complicated. But maybe that's just because I'm struggling to wrap my head around it.